### PR TITLE
Enhance Map Screen Suggestions Menu

### LINKED ISSUE #366: Last suggestion not clickable in Map Screen due to bottom bar android phone

### LINKED ISSUE #393: Search bar in map screen centers around point 0.0 0.0 sometimes
When the search yields only one result, we get centered around point 0.0, 0.0.

### PR DESCRIPTION
**Pull request by Armetiis**
### Description:
This PR introduces several enhancements to the map screen's suggestion menu and resolves two bugs:

1. **#366**
   - The issue was addressed by adding dynamic padding at the bottom of the suggestion menu based on the system's navigation bar insets. This ensures that the last suggestion is fully accessible, preventing it from overlapping with the navigation bar on Android devices.

2. **#393**
   - Previously, the centering of the map was incorrectly based on a "chosen location" state that was not updated correctly. By refactoring the code to ensure that the `handleSuggestionClick` function is called with the correct suggestion coordinates, this bug was resolved. The new implementation ensures that the camera centers accurately on the selected suggestion.

### Enhancements:
- #### UI Improvements:
  - Improved padding within the suggestion menu to enhance spacing and touch targets, ensuring a better user experience.
  - Introduced less opaque dividers between suggestions for improved visual clarity.
  - Utilized surfaces for suggestions to provide a more consistent and modern look.

- #### UX Improvements:
  - Added a drag handle at the top of the suggestion menu, providing users with a visual cue that the menu can be pulled up. This enhances user interaction and makes it more intuitive.

- #### Code Quality Improvements:
  - Modularized the `SuggestionMenu` by creating a separate `handleSuggestionClick` function to avoid code duplication and improve maintainability.
  - Enhanced unique suggestions state management using `distinctBy`, leading to cleaner and more efficient code.
  
  
### Screenshot taken from an Android device with bottom navigation buttons
Before pulling up vs after pulling up

image
image

-- -
Closes #366 
Closes #393

### CODE REVIEW

File: app/src/main/java/com/github/se/cyrcle/ui/map/MapScreen.kt [hunk 1]
```diff
@@ -193,14 +196,11 @@ fun MapScreen(
   val virtualKeyboardManager = LocalSoftwareKeyboardController.current
 
   // List of suggestions from NominatimAPI
-  val listOfSuggestions = addressViewModel.addressList.collectAsState()
+  val listOfSuggestions by addressViewModel.addressList.collectAsState()
```
Comment (LouisJamesVasseur) [inline app/src/main/java/com/github/se/cyrcle/ui/map/MapScreen.kt:199]: val by is indeed better inside of Composable

File: app/src/main/java/com/github/se/cyrcle/ui/map/MapScreen.kt [hunk 2]
```diff
@@ -583,73 +570,94 @@ fun MapScreen(
...
+    suggestion: Address,
+    showSuggestions: MutableState<Boolean>,
+    mapViewportState: MapViewportState,
+    mapViewModel: MapViewModel
+) {
+  showSuggestions.value = false
+
+  mapViewportState.transitionToOverviewState(
+      OverviewViewportStateOptions.Builder()
+          .geometry(mapViewportState.cameraState!!.center)
```
Comment (LouisJamesVasseur) [inline app/src/main/java/com/github/se/cyrcle/ui/map/MapScreen.kt:652]: 100% sure that the cameraState can't be null? (I know it was like this before just ensuring that your new implementation doesn't change things w.r.t that)

Comment (Armetiis) [reply app/src/main/java/com/github/se/cyrcle/ui/map/MapScreen.kt:652]: I don't know. I just took the previous code and put it into a function for modularization purposes. Didn't try to understand what it does in details. Just big picture to add documentation

Comment (Armetiis) [reply app/src/main/java/com/github/se/cyrcle/ui/map/MapScreen.kt:652]: This is something you should maybe discuss with the author of this code and their reviewers?

File: app/src/main/java/com/github/se/cyrcle/ui/map/MapScreen.kt [hunk 3]
```diff
@@ -547,26 +547,13 @@ fun MapScreen(
...
-
-            mapViewportState.setCameraOptions {
-              center(Point.fromLngLat(it.longitude.toDouble(), it.latitude.toDouble()))
-            }
+        if (showSuggestions.value) {
+          if (listOfSuggestions.size > 1) { // If multiple suggestions, show the suggestion menu
+            SuggestionMenu(showSuggestions, listOfSuggestions, mapViewportState, mapViewModel)
+          } else if (listOfSuggestions.size == 1) { // If one suggestion, center the camera on it
+            handleSuggestionClick(
+                listOfSuggestions[0], showSuggestions, mapViewportState, mapViewModel)
```
Comment (andrepeiry) [inline app/src/main/java/com/github/se/cyrcle/ui/map/MapScreen.kt:555]: Nice logic rewriting

**Review by andrepeiry — APPROVED** [review]
Not a lot to say, great refactor, nice modularization and typo fixing !

**Review by LouisJamesVasseur — APPROVED** [review]
A nice sensible rework that works as advertised!

**Review by LouisJamesVasseur — APPROVED** [review]
I'm re-approving as the function still works well from my POV. Good work!

**Pull request merged**